rate.go

```diff
@@ -28,6 +28,11 @@ type Rate struct {
 	EstDeliveryDays        int        `json:"est_delivery_dats,omitempty"`
 }
 
+type Smartrate struct {
+	Rate
+	TimeInTransit map[string]interface{} `json:"time_in_transit,omitempty"`
+}
+
```

Comment:
Go was never really built for inheritance and this type of association usually leads to a lot of complications. For example one would assume that you can instantiate a `Smartrate` struct and initialize the `Rate` attributes as well but one would not be able to to (https://play.golang.org/p/mWvVQ9L2eus).

On the other hand, the use of `map[string]interface{}` to define a well documented json attribute goes against Go and this client library implementation. Technically the `time_in_transit` attribute is part of the `Rate` object according to our documentation: https://www.easypost.com/docs/api#smartrate

Here is what I recommend doing:
* Define `type TimeInTransit struct`
* Add `TimeInTransit` as an attribute under `type Rate struct`
* Debatable, define `type SmartRate Rate` to prettify the client library implementation

Reply:
Ah nice.